### PR TITLE
PRF: Fix Photomod x-dem files georeference

### PR DESCRIPTION
**Pull request by drons**

### CODE REVIEW

File: gdal/frmts/prf/phprfdataset.cpp [hunk 1]
```diff
@@ -608,11 +608,11 @@ GDALDataset* PhPrfDataset::Open( GDALOpenInfo* poOpenInfo )
             abDemMetadataOk[2] && abDemMetadataOk[3] )
         {
             adfGeoTrans[0] = adfDemMetadata[0];
-            adfGeoTrans[1] = (adfDemMetadata[1] - adfDemMetadata[0])/nSizeX;
+            adfGeoTrans[1] = (adfDemMetadata[1] - adfDemMetadata[0])/(nSizeX - 1);
```
Comment (rouault) [inline gdal/frmts/prf/phprfdataset.cpp]: is there a check somewhere that nSizeX != 1 and nSizeY != 1 ?
And generally when you do this, it means that the native geolocation is given with respect to pixel centre. So to go to GDAL convention of pixel-corner, you have to apply a half-pixel offset to gt[0], gt[3]

**Comment by drons** [comment]
> And generally when you do this, it means that the native geolocation is given with respect to pixel centre.

@rouault Thanks for the tip. On this issue, I need to consult with x-dem format developers. And after additional testing, I will let you know.

**Comment by drons** [comment]
@rouault 
We made a detailed comparison of the GDAL and the Photomod, now x-dem geolocation is read in the same way. If you have no objections, then I consider the task done.

**Comment by drons** [comment]
Thanks!

**Pull request merged**